modulefiles/mpi/compilerName/compilerVersion/mpiName/mpiVersion/mapl/mapl.lua

```diff
@@ -0,0 +1,28 @@
+help([[
+]])
+
+local pkgName    = myModuleName()
+local pkgVersion = myModuleVersion()
+local pkgNameVer = myModuleFullName()
+
+local hierA        = hierarchyA(pkgNameVer,2)
+local mpiNameVer   = hierA[1]
+local compNameVer  = hierA[2]
+local mpiNameVerD  = mpiNameVer:gsub("/","-")
+local compNameVerD = compNameVer:gsub("/","-")
+
+conflict(pkgName)
+
+local opt = os.getenv("HPC_OPT") or os.getenv("OPT") or "/opt/modules"
+
+local base = pathJoin(opt,compNameVerD,mpiNameVerD,pkgName,pkgVersion)
+
+setenv("MAPL_ROOT", base)
+prepend_path("CMAKE_MODULE_PATH", pathJoin(base, "share/MAPL/cmake"), ";")
+
+setenv("I_MPI_FABRICS", "shm:ofa")
```

Comment:
Definitely not to be set in the modulefile.  
This should be set where it is used.

Reply:
Addressed in f7aad29cbce68753b99bff1017b6e8ad15de3a12.